viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -66,90 +69,120 @@ int CartoFacade::AddSensorReading(viam_carto_sensor_reading *sr) {
 }  // namespace carto_facade
 }  // namespace viam
 
-extern int viam_carto_init(const viam_carto **ppVC, const viam_carto_config c,
-                           const viam_carto_algo_config ac, char **errmsg) {
+extern int viam_carto_lib_init(viam_carto_lib **ppVCL) {
+    if (ppVCL == nullptr) {
+        return VIAM_CARTO_LIB_INVALID;
+    }
+    if (!((sizeof(float) == 4) && (CHAR_BIT == 8) && (sizeof(int) == 4))) {
```

Comment:
What is this check handling?

Reply:
This is detecting if cartographer is run on a computer with non 32 bit ints & floats & non 8 bit characters & returning an error if so.